### PR TITLE
Add skeleton for cardano-wallet client, with admin status page for demo

### PR DESCRIPTION
**Pull request by ruuda**
## Summary

At some point we will need to integrate with `cardano-wallet` to interact with the blockchain.

 * Add a module for doing this, with one simple query for now.
 * Add an admin status page that displays the results of this query, more for demonstration purposes than anything else

## Details

`cardano-wallet` is written in Haskell and has a Servant API, including a client: https://github.com/input-output-hk/cardano-wallet/blob/3dd2989ab4e1d92f0add996c9e7152e2342d3729/lib/core/src/Cardano/Wallet/Api/Client.hs

However, depending on that library is a pain — it’s not published on Hackage, it uses its own bespoke build system, and it depends on `bech32` which is marked broken in Nixpkgs. I hoped for a moment that I could make it work by just copying the relevant files into our project, but it has enough dependencies that I felt this route was only going to result in misery.

So instead, define types with `FromJSON` instances ourselves, and then use `http-client`. It’s a lot more boilerplate, but it is easy to get it working and it sidesteps a lot of build and dependency problems. Also it allows us to continue using our own types.

For now there is only a function to query the `/network/information` endpoint, and I only parse a subset of the response. This is more of a proof of concept, with this in place it shouldn’t be too difficult to add other GET endpoints.

I split the `Cardano` module for this because I didn’t want to put all of it in one huge file.

## Demo

Visit http://localhost:8080/admin/status

### CODE REVIEW

File: adatipd/cabal.project.local [hunk 1]
```diff
@@ -0,0 +1,6 @@
+ignore-project: False
```
Comment (chloekek) [inline adatipd/cabal.project.local:1]: Missing license identifier.

File: adatipd/src/Adatipd/Cardano/Wallet.hs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    <*> v .: "sync_progress"
+
+makeRequest
+  :: FromJSON a
+  => MonadFail m
+  => MonadIO m
+  => Http.Manager
+  -> Method
+  -> Text
+  -> m a
```
Comment (chloekek) [inline adatipd/src/Adatipd/Cardano/Wallet.hs:82]: Why is this generic over monad? It is only used in `makeRequest'` fixed at type `IO`.

File: adatipd/adatipd.cabal [hunk 3]
```diff
@@ -49,19 +53,26 @@ library
...
         hasql,
+        http-client,
         http-types,
+        mtl,
         optparse-applicative,
         postgresql-binary,
         qrcode-core,
         qrcode-juicypixels,
         scientific,
+        servant,
```
Comment (chloekek) [inline adatipd/adatipd.cabal:71]: This isn’t listed in shell.nix. And I don’t think we need it.

File: adatipd/adatipd.cabal [hunk 4]
```diff
@@ -49,19 +53,26 @@ library
         Adatipd.Web.NotFound
 
     build-depends:
+        aeson,
         base,
         blaze-html,
         blaze-markup,
         bytestring,
+        containers,
```
Comment (chloekek) [inline adatipd/adatipd.cabal:61]: Also seems unused.

File: adatipd/adatipd.cabal [hunk 5]
```diff
@@ -49,19 +53,26 @@ library
...
         optparse-applicative,
         postgresql-binary,
         qrcode-core,
         qrcode-juicypixels,
         scientific,
+        servant,
         text,
         time,
+        transformers,
+        unordered-containers,
```
Comment (chloekek) [inline adatipd/adatipd.cabal:75]: Also seems unused.

File: adatipd/src/Adatipd/Cardano/Wallet.hs [hunk 6]
```diff
@@ -0,0 +1,109 @@
+-- SPDX-License-Identifier: AGPL-3.0-only
+
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE FlexibleContexts #-}
+
+module Adatipd.Cardano.Wallet
```
Comment (chloekek) [inline adatipd/src/Adatipd/Cardano/Wallet.hs:6]: Would be nice to add documentation with some context about cardano-wallet and why we have our own module.

**Review by chloekek — CHANGES_REQUESTED** [review]
Pretty nice. I also like the status page. There are some leftovers in the code:

File: adatipd/cabal.project.local (hunk 1 above)
Comment (ruuda) [reply adatipd/cabal.project.local:1]: Fixed.

File: adatipd/src/Adatipd/Cardano/Wallet.hs (hunk 2 above)
Comment (ruuda) [reply adatipd/src/Adatipd/Cardano/Wallet.hs:82]: I was playing around a bit with the types, at first I considered making a `MonadWallet` typeclass, but then I settled on just the concrete `WalletIO`. Will remove the premature generalization.

File: adatipd/adatipd.cabal (hunk 3 above)
Comment (ruuda) [reply adatipd/adatipd.cabal:71]: This was a leftover from me trying to get `cardano-wallet`’s client to work. Removed now.

File: adatipd/adatipd.cabal (hunk 4 above)
Comment (ruuda) [reply adatipd/adatipd.cabal:61]: Removed. I expect it to come back at some point though :sweat_smile:

File: adatipd/adatipd.cabal (hunk 5 above)
Comment (ruuda) [reply adatipd/adatipd.cabal:75]: It was almost used, an `Aeson.Object` is a `HashMap Text Value`, and I briefly used `lookup`, but now it is no longer required. Removed.

File: adatipd/src/Adatipd/Cardano/Wallet.hs (hunk 6 above)
Comment (ruuda) [reply adatipd/src/Adatipd/Cardano/Wallet.hs:6]: Added.

**Comment by ruuda** [comment]
Also added a third schema example and a handler for it; apparently the network tip can be missing.

**Review by chloekek — APPROVED** [review]
:shipit:

**Pull request merged**